File: Loom.Core/Models/ToolCallInvocation.vb
Namespace Models
    ''' <summary>
    ''' Properties of tool when invoked
    ''' </summary>
    Public Class ToolCallInvocation
        Public Property Type As String = "function_call"
        Public Property CallId As String
        Public Property ToolName As String
        Public Property Arguments As Dictionary(Of String, Object)
    End Class

End Namespace

File: implementations/csharp/src/Loom.Core/Models/ToolCallInvocation.cs
using System.Collections.Generic;

namespace Loom.Core.Models
{
    /// <summary>
    /// Properties of tool when invoked
    /// </summary>
    public class ToolCallInvocation
    {
        public string Type { get; set; } = "function_call";
        public string CallId { get; set; }
        public string ToolName { get; set; }
        public Dictionary<string, object> Arguments { get; set; }
    }
}
